docs/design-documents/hal/0005-pin-names-Arduino-Uno-standard.md

```diff
@@ -116,15 +115,53 @@ Some Dx pinnames may support the usage of PWM or Timers functions. Although this
 
 Note this might be one of the main differencess accross Mbed boards and therefore the application should not assume the same behaviour for PWM and Timers for them.
 
-The RESET signal should be defined in the Arduino Uno connector to let the user or an external component to put the MCU in reset state.
+The RESET signal should be defined in the Arduino Uno connector to let the user or an external component put the MCU in reset state.
 
-The VIN signal isn't strictly defined in the Arduino standard in some cases may work as a bi-directional power supply. Tipically it could accept between 5 to 12 volts, but it's not a requirement.
+The VIN signal isn't strictly defined in the Arduino standard and in some cases may work as a bi-directional power supply. Tipically it could accept between 5 to 12 volts, but it's not a requirement.
 
+### Usage of Arduino Uno pin-names in an application
 
-**Using Arduino Uno pins from an application**
+**Configuration**
 
-> Pending to add details
+The pin names defined to each of the MCUs in PinNames.h are considered the default configuration for every target in Mbed OS.
+They are picked automatically at compile time and can used in the application.
 
+However, it's possible to override the default configuration. This is the list of configuration files and a description of each of them:
+
+- `PinNames.h`: includes the default configuration for Arduino Uno pins and how they are connected to the MCU as described in this document.
+- `mbed_lib.json`: configuration for specific components to use the default Arduino Uno pin-names or select other MCU pins if required. This file can be defined in Mbed OS or in an external library for the component, for [example](https://github.com/ARMmbed/mbed-os/blob/master/storage/blockdevice/COMPONENT_SD/mbed_lib.json).
+- `mbed_app.json`: application configuration that inherits the `mbed_lib.json` if exists and can override the pin configuration for the component if required. This file is usually placed at the root directory of the application.
+- main application: the `.cpp` application could initialize the drivers with the default Arduino Uno pin-name configution, use the configuration in `mbed_app.json`, or use custom pins if required.
+
+**Examples of usage of Arduino Uno pin names**
+
+This section provides a few examples with guidelines on using Arduino Uno pin names. It's not a guide to use Mbed Drivers API (please refer to the [documentation](https://os.mbed.com/docs/mbed-os/latest/apis/drivers.html) for details).
+
+It's possible to use digital signals in both Dx and Ax pins:
+
+    DigitalOut myGPIO1(ARDUINO_UNO_D3); // Output signal
+    DigitalOut myGPIO2(ARDUINO_UNO_D4); // Input signal
+    DigitalOut myGPIO3(ARDUINO_UNO_A0); // A0 can be used as digital signal as well
+
+Usage of ADC is only possible on Ax pins:
```

Comment:
"only" is maybe not true.
Nothing could prevent vendor to connect a ADC capable pin on Dx... :-)

Reply:
In terms of Arduino Uno header which this document is dealing with ADC's can only be used on the A0-A5 pins. 
If another pin on the Arduino Uno header is capable of being an ADC then it must be initialised using the board's processor pin(Px_xx) connected to that Arduino Uno header pin and not using the Arduino Uno pin alias

Reply:
I tend to agree with both comments but also think we should differentiate on minimum requirements for official Arduino Uno connector and official applications. Hence I propose the following:
- The analog input signals in Arduino Uno connector must supported on at least the Ax pins.
- The MCU may support the usage of ADC functions in other non-Ax pins. This is something that developers could use in their own custom applications. It's recommended they use the MCU's pin names as shown here:
    `AnalogIn myADC2(P5_23);             // Custom usage of ADC`

Let me know what you think - I've just pushed an update